src/reducers/ServerConfigState.ts

```diff
@@ -0,0 +1,21 @@
+import { ServerConfig } from '../store/Store';
+import { KialiAppAction } from '../actions/KialiAppAction';
+import { getType } from 'typesafe-actions';
+import { ServerConfigActions } from '../actions/ServerConfigActions';
+
+export const INITIAL_SERVER_CONFIG: ServerConfig | null = null;
+
+// This Reducer allows changes to the 'serverConfig' portion of Redux Store
+const serverConfig = (
+  state: ServerConfig | null = INITIAL_SERVER_CONFIG,
+  action: KialiAppAction
+): ServerConfig | null => {
+  switch (action.type) {
+    case getType(ServerConfigActions.setServerConfig):
+      return action.payload;
```

Comment:
I think it's worth to check if the `duration` stored in redux is still valid and update the store if it no longer will be a valid duration. That way, components won't try to set `undefined` values.